File: DTOs/Sources/RideServiceDTOs/GetEventParticipationDTO.swift
import Foundation

public struct GetEventParticipationDTO: Codable {
    public var id: UUID
    public var name: String
    public var itsMe: Bool
    public var participates: UsersParticipationState
    
    public init(id: UUID, name: String, itsMe: Bool, participates: UsersParticipationState) {
        self.id = id
        self.name = name
        self.itsMe = itsMe
        self.participates = participates
    }
}

public enum UsersParticipationState: String, Codable, CaseIterable {
    case nothing
    case absent
    case present
}

File: android/app/src/main/java/net/ipv64/kivop/dtos/RideServiceDTOs/GetEventParticipationDTO.kt
package net.ipv64.kivop.dtos.RideServiceDTOs

import java.util.UUID
import java.time.LocalDateTime

data class GetEventParticipationDTO (
    var id : UUID,
    var name : String,
    var itsMe : Boolean,
    var participates : UsersParticipationState,
)
enum class UsersParticipationState {
    nothing,
    absent,
    present,
}
